pages/api/interest-zones/[id].ts

```diff
@@ -0,0 +1,25 @@
+import ZoneSchema from '../../../data/zone.schema';
+import { dbConnection } from '../../../services/data.service';
+
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+const databaseConnection = dbConnection;
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<any>) {
+    switch (req.method) {
+        case 'GET':
+            try{
+                const interestZone = await ZoneSchema.findById(req.query.id).exec();
+        
+                if (!interestZone) {
+                    throw new Error(`Zone with id ${req.query.id} not found`);
+                }
```

Comment:
Consider adding a ZoneService, and move the three rows above to a dedicated method. Ideally, this `try` section will include only two rows, a service call and the `res.json()` call